devito/core/gpu.py

```diff
@@ -176,6 +191,10 @@ def _specialize_clusters(cls, clusters, **kwargs):
         # Reduce flops
         clusters = cse(clusters, sregistry)
 
+        # Blocking to define thread blocks
+        if options['blocklazy']:
+            clusters = blocking(clusters, options)
```

Comment:
Still think would be nice to have this cpu/gpu core [art merged as it is very similar but I understand may be a pain.

Reply:
Agreed. I think we're getting there, slowly, PR after PR. With this PR for example we're dropping the explicit `advanced-fsg` pipeline. In the past we've dropped others.

Reply:
Right...theoretically as far as we move to exploring the optimization order impact we are dropping pipelines...theoretically always,...